### ISSUE TITLE
Python 3: error on 'List histories' view

### ISSUE THREAD

**Issue by mauritsvanrees** [collective/collective.revisionmanager#25]
```
http://localhost:8080/nl/@@histories
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 162, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 359, in publish_module
  Module ZPublisher.WSGIPublisher, line 262, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 63, in call_object
  Module collective.revisionmanager.browser.views, line 135, in __call__
  Module Products.Five.browser.pagetemplatefile, line 126, in __call__
  Module Products.Five.browser.pagetemplatefile, line 61, in __call__
  Module zope.pagetemplate.pagetemplate, line 135, in pt_render
  Module Products.PageTemplates.engine, line 367, in __call__
  Module z3c.pt.pagetemplate, line 176, in render
  Module chameleon.zpt.template, line 307, in render
  Module chameleon.template, line 214, in render
  Module chameleon.utils, line 75, in raise_with_traceback
  Module chameleon.template, line 192, in render
  Module 21f470d82f5458867c2eae963a1bb084, line 1374, in render
  Module 00481695307195eacae4764382b2320e, line 382, in render_master
  Module 0d155c43d04045902d5ac6dd9a7b0aa2, line 871, in render_master
  Module 00481695307195eacae4764382b2320e, line 365, in __fill_content
  Module 21f470d82f5458867c2eae963a1bb084, line 1352, in __fill_prefs_configlet_content
  Module 21f470d82f5458867c2eae963a1bb084, line 652, in render_master
  Module zope.tales.pythonexpr, line 73, in __call__
   - __traceback_info__: (view.humanize_size(histData['size']))
  Module <string>, line 1, in <module>
  Module collective.revisionmanager.browser.views, line 149, in humanize_size
TypeError: object of type 'zip' has no len()
```

Looks like this will be fixed in PR #18.
I just wanted to quickly confirm that this is a problem.

**Comment by mauritsvanrees** [comment]
I am looking into this, and will update the `python3` branch. When using this branch, the above is fixed, but then I get:

    TypeError: '>' not supported between instances of 'method' and 'int'

caused by this template expression:

    repeat['wcinfo'].length > 1

I have seen the same in customer code. On Python 2, comparing `method > 1` always return True, and on Python 3 you get an error. Maybe here in the case of Python 2 the `length` was a property and this has changed; I have not checked yet.

This goes wrong in the tests too.

I tried fixing this by splitting it up:

    <br tal:define="length repeat/wcinfo/length"
           tal:condition="python:length > 1"/>

but that gives:

    TypeError: object of type 'list_iterator' has no len()

The repeat syntax may have changed a bit in Plone 5.2.

**Comment by mauritsvanrees** [comment]
Got it. The Zope docs have a section on [repeat variables](https://zope.readthedocs.io/en/latest/zopebook/AdvZPT.html#repeat-variables). It says about `length`:

> length - length of the sequence, which will be the total number of repetitions. Unsafe - see warning below.

And the warning is:

> The length attribute may not be available and lead to an error for sequences that don’t support the `__len__` method. To get around that you can use tal:define to capture the sequence you’re iterating over first and then determining its length by calling e.g. `tal:define="seq_len python: len(mysequence)"`

Doing it like this works:

```
      <tal:wc define="wcinfos python:list(histData['wcinfos']);
                        multi python:len(wcinfos) > 1">
        <td>
          <tal:wcinfos repeat="wcinfo wcinfos">
            <tal:portaltype content="wcinfo/portal_type">portal_type</tal:portaltype>
            <br tal:condition="multi"/>
          </tal:wcinfos>
        </td>
      </tal:wc>
```

**Comment by mauritsvanrees** [comment]
Fixed with the merged PR #18. Now a release would be good.
